File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Action/Animations/ShoutAction.ts
import DungeonManager from "../../DungeonManager";
import { MegaSpeechBubble } from "../../Effect/Combat/MegaSpeechBubble";
import AbstractCreature from "../../Unit/AbstractCreature";
import { AbstractPlayer } from "../../Unit/Character/AbstractPlayer";
import AbstractGameAction, { ActionType } from "../AbstractGameAction";

export class ShoutAction extends AbstractGameAction{
    Type = ActionType.TEXT;
    public constructor(private _Source:AbstractCreature,private _Text:string,private _Duration:number = 0.5,private _BubbleDuration:number = 3) {
        super();
    }

    async Update(): Promise<boolean> {
        let hitBox = this._Source.GetHitBox();
        DungeonManager.EffectManager.Play(new MegaSpeechBubble(hitBox.CX, hitBox.CY,this._Text,this._BubbleDuration,!(this._Source instanceof AbstractPlayer)));
        return super.Update();
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Action/Combat/ShoutAction.cs
using Cysharp.Threading.Tasks;
using ModTheSpire.Dungeon;
using ModTheSpire.Effect.Combat;
using ModTheSpire.Event;
using ModTheSpire.Unit;
using ModTheSpire.Unit.Character;

namespace ModTheSpire.Effect.Other
{
    public class ShoutAction :  IGameAction
    {
        public float Duration { get; set; }
        public ActionType Type { get; } = ActionType.TEXT;
        private AbstractCreature mSource;
        private string mText;
        private int mInterval;
        private int mBubbleDuration;

        public ShoutAction(AbstractCreature source, string text, int interval = 500, int bubbleDuration = 3000)
        {
            mSource = source;
            mText = text;
            mInterval = interval;
            mBubbleDuration = bubbleDuration;
        }

        public void Init()
        {
        }
        
        public UniTask<bool> Update()
        {
            var hitBox = this.mSource.DisplayObject.GetHitBox();
            DungeonManager.EffectManager.Play(new MegaSpeechBubble(hitBox.center.x, hitBox.center.y, mText, mBubbleDuration,
                !(mSource is AbstractPlayer)));
            return UniTask.FromResult(true);
        }
    }
}